### ISSUE TITLE
crash: Unhandled invalid instruction at 0x8b772008

### ISSUE THREAD

**Issue by BPaden** [inolen/redream#105]
Ran into this in the middle of playing Spider-Man. For #37, the game appears to be in a playable state otherwise. For convenience I cut the retroarch init stuff out since it appears to be an emulation problem, but here's a log.

```
boot_load_rom loaded '/data/roms/bios/boot.bin'
pvr_reconfigure_spg mode=VGA pixel_clock=13500000 line_clock=15734 vcount=262 hcount=857 interlace=0 vbstart=260 vbend=336
loading /data/roms/dreamcast/spider-man/Spider-Man (USA).gdi
gdi_parse track=1 filename='/data/roms/dreamcast/spider-man/track01.bin' fad=150 secsz=2352
gdi_parse track=2 filename='/data/roms/dreamcast/spider-man/track02.raw' fad=870 secsz=2352
gdi_parse track=3 filename='/data/roms/dreamcast/spider-man/track03.bin' fad=45150 secsz=2352
gdi_parse track=4 filename='/data/roms/dreamcast/spider-man/track04.raw' fad=159130 secsz=2352
gdi_parse track=5 filename='/data/roms/dreamcast/spider-man/track05.bin' fad=159701 secsz=2352
gdrom_set_disc SPIDER-MAN V1.000 - T13008N
sh4_ccn_reset
pvr_reconfigure_spg mode=VGA pixel_clock=13500000 line_clock=15734 vcount=262 hcount=857 interlace=0 vbstart=260 vbend=336
pvr_reconfigure_spg mode=NTSC pixel_clock=13500000 line_clock=31468 vcount=524 hcount=857 interlace=1 vbstart=260 vbend=336
arm7_reset
sh4_ccn_reset
pvr_reconfigure_spg mode=NTSC pixel_clock=13500000 line_clock=31468 vcount=524 hcount=857 interlace=1 vbstart=516 vbend=36
arm7_reset
pvr_reconfigure_spg mode=NTSC pixel_clock=13500000 line_clock=31468 vcount=524 hcount=857 interlace=1 vbstart=516 vbend=36
sh4_ccn_reset
pvr_reconfigure_spg mode=NTSC pixel_clock=13500000 line_clock=31468 vcount=524 hcount=857 interlace=1 vbstart=516 vbend=36
pvr_reconfigure_spg mode=NTSC pixel_clock=13500000 line_clock=31468 vcount=524 hcount=857 interlace=1 vbstart=516 vbend=36
sh4_ccn_reset
pvr_reconfigure_spg mode=NTSC pixel_clock=13500000 line_clock=31468 vcount=524 hcount=857 interlace=1 vbstart=516 vbend=36
arm7_reset
pvr_reconfigure_spg mode=NTSC pixel_clock=13500000 line_clock=31468 vcount=524 hcount=857 interlace=1 vbstart=516 vbend=36
sh4_ccn_reset
backend overflow, resetting code cache
sh4_ccn_reset
backend overflow, resetting code cache
sh4_ccn_reset
unexpected write to 0x00000010
unexpected read from 0x00000008
Unhandled invalid instruction at 0x8b772008
```

**Comment by inolen** [comment]
Hmph, my guess would be these may be a self-modifying code issue.

The only way to validate that would be to play through with the interpreter as well and see if it crashes in the same spot. Currently there's not a command-line option to force that, but you can do it by replacing `ARCH_X64` with `0` in https://github.com/inolen/redream/blob/master/src/guest/sh4/sh4.c.

**Comment by BPaden** [comment]
Reproducing (or not) will be a little difficult because it happened randomly during play and not at a specific spot, but later I'll recompile redream without the dynarec and see if I get anything approaching usable FPS (I say as I eye my computer with an old core 2 from 2009 with great skepticism)

**Comment by inolen** [comment]
@BPaden one interesting note - this is the exact same sequence of invalid addresses that Red Dog gets now on startup.

Were you ever able to repro this? I'm going to try and fix Red Dog now.

**Comment by BPaden** [comment]
I haven't seen this again, but I haven't spent too much time trying to reproduce it. I was going to wait for save states.

I'm fine with assuming fixing Red Dog will fix this as well. If it pops up again, it can always be reopened.

**Comment by inolen** [comment]
Should be closed with 3c51b945ba63d0c0c49df610f93e647c2443f620, but please let me know if it ever shows back up.
